bpf/lib/nat.h

```diff
@@ -827,9 +827,19 @@ snat_v4_nat(struct __ctx_buff *ctx, struct ipv4_ct_tuple *tuple,
 		case ICMP_ECHOREPLY:
 			return NAT_PUNT_TO_STACK;
 		case ICMP_DEST_UNREACH:
-			if (icmphdr.code != ICMP_FRAG_NEEDED)
+			switch (icmphdr.code) {
+			case ICMP_NET_UNREACH:
+			case ICMP_HOST_UNREACH:
+			case ICMP_PROT_UNREACH:
+			case ICMP_PORT_UNREACH:
+			case ICMP_FRAG_NEEDED:
```

Comment:
according to the original rfc there's also a 6th code, `ICMP_SR_FAILED`, and then some newer codes got introduced with more recent rfcs (https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml#icmp-parameters-codes-3). Is there a reason for allowing only a subset of them?

Reply:
Not really - I was mostly interested in `ICMP_PORT_UNREACH` (see the linked GH issue), without going down the rabbit hole of implementing the whole spec. But expected that checking a range of values will make it easier for the compiler / verifier.

We can easily add more code points in a follow-up PR, if you want to complete the picture (or see any specific ones that are widely used). I wanted to tackle `TIME_EXCEEDED` next anyway.

Reply:
ack, my thinking was that this could have been simplified to something like:
```c
if (icmphdr.code > 15)
    return DROP_UNKNOWN_ICMP_CODE;
```
in order to avoid listing each individual code, while increasing coverage, but fine to keep that as a followup if the aim is to support `ICMP_PORT_UNREACH` :+1: